velox/exec/RowContainer.cpp

```diff
@@ -1275,36 +1275,47 @@ RowComparator::RowComparator(
   }
 }
 
-bool RowComparator::operator()(const char* lhs, const char* rhs) {
+int32_t RowComparator::compare(const char* lhs, const char* rhs) {
   if (lhs == rhs) {
-    return false;
+    return 0;
   }
   for (auto& key : keyInfo_) {
     if (auto result = rowContainer_->compare(
             lhs,
             rhs,
             key.first,
             {key.second.isNullsFirst(), key.second.isAscending(), false})) {
-      return result < 0;
+      return result;
     }
   }
-  return false;
+  return 0;
 }
 
-bool RowComparator::operator()(
+bool RowComparator::operator()(const char* lhs, const char* rhs) {
+  return compare(lhs, rhs) <= 0;
+}
+
+int32_t RowComparator::compare(
+    const char* lhs,
     const std::vector<DecodedVector>& decodedVectors,
-    vector_size_t index,
-    const char* rhs) {
+    vector_size_t index) {
```

Comment:
Can we not continue with the existing API? It seems less risky not to change.
``` 
    const std::vector<DecodedVector>& decodedVectors,
      vector_size_t index,
      const char* rhs);
```

Reply:
In the original PR I had retained the current API.

The usage of the current API is a bit awkward (as rhs is used as lhs in the comparison) so its counterintuitive, and I also changed the way comparisons were done to make it easier to use. This came as a result of a review comment.

The code is used only in TopN and TopNRowNumber exercised by Fuzzer.

Its definitely less risky to keep the original API. And I will cheange to it.